SlackLogger/LoggerProvider.cs

```diff
@@ -13,15 +12,14 @@ internal class LoggerProvider : ILoggerProvider
         private readonly SlackLoggerOptions _options;
         private readonly ScopeSettings _scopeSettings;
 
-        public LoggerProvider(IOptions<LoggerFilterOptions> filterOptions, IOptions<SlackLoggerOptions> options, IConfiguration configuration, IHostingEnvironment env)
+        public LoggerProvider(IOptions<LoggerFilterOptions> filterOptions, IOptions<SlackLoggerOptions> options, IConfiguration configuration)
         {
             _options = options.Value;
             _options.Merge(configuration.GetSection("Logging:Slack"));
             
-            _options.EnvironmentName = env?.EnvironmentName;
-            _options.ApplicationName = !string.IsNullOrEmpty(_options.ApplicationName)
-                ? _options.ApplicationName
-                : env?.ApplicationName;
+            _options.ApplicationName = String.IsNullOrEmpty(_options.ApplicationName)
+                ? System.Reflection.Assembly.GetEntryAssembly()?.GetName()?.Name
```

Comment:
I think using reflection here will work most of the time but I'd like to test it out and see what it does in my cases to make sure it will behave the way we would expect

Reply:
Published `3.0.0-rc1` if you wanna test

Reply:
Currently running 3.0.0-rc1 on our staging environment for both dotnet core app and aspnet core. So far so good

Reply:
Tested with reflection on windows and in a linux docker container 👍